veps/sig-storage/dic-on-heterogeneous-cluster/dic-on-heterogeneous-cluster.md

```diff
@@ -0,0 +1,459 @@
+`spec` field is CDI's `DataImportCronSpec`, that one of its nested fields is the `DataVolumeSourceRegistry` type.
+
+In addition, the SSP `spec` field will be extended with the new `cluster` with a new struct type, similar to:
+
+```go
+type Cluster struct {
+//WorkloadArchitectures is a list of the architectures supported by the cluster
+// +optional
+WorkloadArchitectures []string `json:"workloadArchitectures,omitempty"`
+}
+```
+
+### HyperConverged API Changes
+By adopting the new CDI and SSP APIs, the `DataImportCronTemplate` type in HyperConverged will include the CDI change,
+as the type of the `spec` field is CDI's `DataImportCronSpec`, that one of its nested fields is the
+`DataVolumeSourceRegistry` type.
+
+<!--
+Tangible API examples used for discussion
+-->
+
+## Alternatives
+
+HCO could create a new `DataImportCronTemplate` object for each architecture in the `SSP` CR, and the user will be
+responsible for.
+
+pros:
+
+- no changes are required from SSP, except for adopting the new CDI API.
+
+cons:
+
+- the user will be responsible for creating and maintaining the `DataImportCronTemplate` objects, and
+  the user will need to create a new `DataImportCronTemplate` object for each architecture.
+- the SSP CR will be huge, and will contain a lot of `DataImportCronTemplate` objects. It will be hard to maintain it
+  or understand it.
+- it looses the meaning of the templating mechanism, by the `DataImportCronTemplate` type.
+
+## Scalability
+
+n/a
+
+## Update/Rollback Compatibility
+
+### Upgrade Compatibility
+As outlined above, the SSP operator should maintain the current behavior for `DataImportCronTemplate` objects, that
+do not include the `ssp.kubevirt.io/dict.architectures` annotation. Similarly, the cdi-importer preserve its existing
+behavior when the `DataSource`'s `spec.template.spec.source.registry.platform.architecture` is not populate.
+
+his ensures forward compatibility for all`DataImportCronTemplates` objects. For instance, even after
+upgrading the HCO to a version that implements this feature, the custom `DataImportCronTemplate` objects will remain
+unaltered and continue to function as expected, allowing users to update them at their own pace.
+
+### Rollback Compatibility
+For the same reason, an older version of the HCO will reconcile the `DataImportCronTemplate` objects in the `SSP` CR
+without including the `spec.workloads.nodePlacement` field and the `ssp.kubevirt.io/dict.architectures` annotation.
+The SSP operator will treat these as legacy `DataImportCronTemplate` objects, as described above.
+
+The SSP operator is responsible for removing any `DataImportCron` CRs created by the SSP that are not defined by any
+current `DataImportCronTemplate` object in the SSP CR.
+
+## Functional Testing Approach
+Effective functional tests will require to build a test environment with heterogeneous cluster, which presents a
```

Comment:
Depending on how CDI implements the import feature we could import multiple arches on a single arch cluster

Reply:
That will work for ssp and cdi test, but not for e2e with hco.